src/components/colors/colors.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @ngdoc module
+ * @name material.components.colors
+ *
+ * @description
+ * Colors directive
+ */
+angular.module('material.components.colors', [
+    'material.core'
+  ])
+  .directive('mdColors', mdColorsDirective);
+
+/**
+ * @ngdoc directive
+ * @name mdColors
+ * @module material.components.colors
+ *
+ * @restrict A
+ *
+ * @description
+ * `mdColors` directive will apply the requested color from the palette as inline style on its element,
+ * The format will be similar to our color defining in the scss files:
+ * ## ‘[?theme]-[palette]-[?hue]-[?opacity]’
+ * - [theme]    - default value is the default theme
+ * - [palette]	- can be either palette name or primary/accent/warn/background
+ * - [hue]		  - default is 500
+ * - [opacity]	- default is 1
+ * - ? 		      - optional
+ *
+ * @usage
+ * <hljs lang="html">
+ *   <div md-colors="{background: 'myTheme-accent-900-0.43'}">
+ *     <div md-colors="{color: 'red-A100', border-color: 'primary-600'}">
+ *       <span>Color demo</span>
+ *     </div>
+ *   </div>
+ * </hljs>
+ *
+ */
+
+function mdColorsDirective($mdColorPalette, $mdTheming) {
+  return {
+    restrict: 'A',
+    link: function (scope, elem, attrs) {
+      scope.$watch(attrs.mdColors, function (val) {
+        Object.keys(val).forEach(function (key) {
```

Comment:
You could spare calling `Object.keys` and iterating it by using `angular.forEach(function(value, key){ ... });`